console/src/main/scripts/plugins/metrics/ts/explorer.ts

```diff
@@ -0,0 +1,282 @@
+        this.selectedMetric['resource'] = _.cloneDeep(this.selectedResource);
+        this.selectedMetric['feed'] = _.cloneDeep(this.selectedFeed);
+        this.addNewChartToController(this.selectedMetric);
+        this.addMetricToStorage();
+        this.refresh();
+      }
+    }
+
+    private addNewChartToController(metric) {
+      this.charts.push(metric);
+      this.chartType[metric.id] = metric.type.type;
+      this.chartUnit[metric.id] = metric.type.unit;
+    };
+
+    public removeChart(chart) {
+      this.$log.log('Remove ' + chart);
+      let index = this.charts.indexOf(chart,0);
+      if (index > -1) {
+        this.charts.splice(index,1);
+        let id = chart.id;
+        index = this.chartData.indexOf(id,0);
+        if (index > -1) {
+          this.chartData.splice(index,1);
+        }
+        this.refresh();
+      }
+    }
+
+    public refresh() {
+      this.alertList = [];
+      this.getAlerts();
+      this.getMetricData();
+    }
+
+    public getMetricData() {
+      _.forEach(this.charts, (res: any) => {
+        if (res.type.type === 'GAUGE') {
+          this.MetricsService.retrieveGaugeMetrics(this.$rootScope.currentPersona.id,
+            res.id,
+            this.startTimeStamp,
+            this.endTimeStamp,
+            60
+          ).then((data) => {
+            this.formatChartGaugeData(res.id, data);
+          });
+        } else if (res.type.type === 'COUNTER') {
+          this.MetricsService.retrieveCounterMetrics(this.$rootScope.currentPersona.id,
+            res.id,
+            this.startTimeStamp,
+            this.endTimeStamp,
+            60
+          ).then((data) => {
+            this.formatChartData(res.id, data);
+          });
+        } else if (res.type.type === 'AVAILABILITY') {
+          this.MetricsService.retrieveAvailabilityMetrics(this.$rootScope.currentPersona.id,
+            res.id,
+            this.startTimeStamp,
+            this.endTimeStamp,
+            60
+          ).then((data) => {
+            this.formatChartData(res.id, data);
+          });
+        } else {
+          this.$log.log('Unknown type ' + res.type.type);
+        }
+      });
+    }
+
+    private formatChartGaugeData(resourceId, data) {
+      if (data.length) {
+        let scale = 1 /  MetricsService.getMultiplier(data);
+        this.chartData[resourceId] = MetricsService.formatBucketedChartOutput(data,scale);
+      }
+    }
+
+    private formatChartData(resourceId, data) {
+      if (data.length) {
+        this.chartData[resourceId] = MetricsService.formatBucketedChartOutput(data);
+      }
+    }
+
+    private addMetricToStorage(): void {
+        this.$log.log('addMetricToStorage');
+      this.$localStorage.hawkular_charts = this.charts;
+    }
+  }
+
+  //_module.config(['ngClipProvider', (ngClipProvider) => {
+  //  ngClipProvider.setConfig({
+  //    zIndex: 50
+  //  });
+  //}]);
+
+  _module.controller('ExplorerController', ExplorerController);
```

Comment:
Ideally, a name like `InventoryExplorerController` is more descriptive than `ExplorerController`